spectator-ext-ipc/src/main/java/com/netflix/spectator/ipc/IpcTagKey.java

```diff
@@ -163,6 +163,13 @@
    * The port number connected to on the server.
    */
   serverPort("ipc.server.port"),
+  
+  /**
+   * Indicates that an artificial failure was injected into the request processing for testing purposes. 
+   * The outcome of that failure will be reflected in the other error tags. 
+   * Valid values are {@code none}, {@code failure}, {@code delay}.
```

Comment:
If the set of values is fixed, should there be a dedicated enum for that type that can be used? Similar to IpcResult or IpcStatus:

https://github.com/Netflix/spectator/blob/master/spectator-ext-ipc/src/main/java/com/netflix/spectator/ipc/IpcResult.java

This would also allow us to integrate checks for this into the validation to help better ensure consistency going forward:

https://github.com/Netflix/spectator/blob/master/spectator-ext-ipc/src/main/java/com/netflix/spectator/ipc/IpcMetric.java#L182

Reply:
Ahh, yeah, that could be useful. Will make the updates -- thanks!

Reply:
I moved the valid values into an enum, enabled validation, and added a couple of tests.